### ISSUE TITLE
Why is the Trainable parameter false for all layers in yolov8n.pt?

### ISSUE THREAD

**Issue by webzhao9125** [ultralytics/ultralytics#14907]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I ran the code below and looked at the Trainable parameter for each layer in yolov8n.pt. I found that the Trainable of all the layers is False, if all of them are False, wouldn't the loss back propagation be useless during training？

```python
from ultralytics import YOLO

model = YOLO('yolov8n.pt') 


for name, param in model.named_parameters():
    print(f"Layer: {name}, Trainable: {param.requires_grad}")
```

![image](https://github.com/user-attachments/assets/9d64e1d8-75ee-43a6-935d-a2b42a853c73)

**Comment by pderrenger** [comment]
@webzhao9125 it sounds like the parameters might not be set to require gradients, which is essential for backpropagation during training. This could be due to the model being in evaluation mode. Ensure that the model is set to training mode by using `model.train()` before checking the `requires_grad` attribute. If the issue persists, please verify that you are using the latest version of the Ultralytics package, as updates may resolve such issues. If you continue to encounter problems, feel free to provide more details.

**Comment by Y-T-G** [comment]
It's disabled so that you don't have to do `torch.no_grad()` redundantly. It gets enabled before training.

**Comment by webzhao9125** [comment]
OK I got it, thanks!

**Comment by pderrenger** [comment]
You're welcome! If you have any more questions or need further assistance, feel free to ask. Happy training!

**Comment by glakhanp** [comment]
How can I enable parameters to require grad?

**Comment by Y-T-G** [comment]
https://github.com/ultralytics/ultralytics/issues/10071#issuecomment-2440038234
